### PR TITLE
Day Label Formatter

### PR DESCRIPTION
**Pull request by dandc87**
Completes the label customization ability within the widget

### CODE REVIEW

File: library/src/main/java/com/prolificinteractive/materialcalendarview/format/DateFormatDayFormatter.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ */
+public class DateFormatDayFormatter implements DayFormatter {
+
+    private final DateFormat dateFormat;
+
+    /**
+     * Uses "MMMM yyyy" for formatting
+     */
+    public DateFormatDayFormatter() {
+        this.dateFormat = new SimpleDateFormat("d", Locale.getDefault());
```
Comment (ireneduke) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/format/DateFormatDayFormatter.java]: What is the "d" for?

Comment (dandc87) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/format/DateFormatDayFormatter.java]: Single "d" will create the shortest day of the month string. I should probably fix the javadoc for this constructor though.

File: library/src/main/java/com/prolificinteractive/materialcalendarview/format/DateFormatDayFormatter.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+    /**
+     * @param format the format to use
+     */
+    public DateFormatDayFormatter(DateFormat format) {
+        this.dateFormat = format;
+    }
+
+    @Override
+    public String format(CalendarDay day) {
```
Comment (chahinem) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/format/DateFormatDayFormatter.java]: the case when `dateFormat` is null is not handled. Can it be null ?

Comment (dandc87) [inline library/src/main/java/com/prolificinteractive/materialcalendarview/format/DateFormatDayFormatter.java]: Added @NonNull annotations

**Pull request merged**